decentraland/helpers/index.js

```diff
@@ -178,6 +188,46 @@ export const openURL = async url => {
   }
 };
 
+export const restoreCreationStateOfAccountFromBlockchain = async account => {
+  const provider = ProviderFactory.getProvider();
+  const { address } = account;
+
+  const contracts = getContracts();
+  const { marketplaceContract, manaContract } = contracts;
+
+  const ethersBalance = await provider.getBalance(address);
+  const manaBalance = await manaContract.balanceOf(address);
+  const allowance = await manaContract.allowance(
+    address,
+    marketplaceContract.getAddress()
+  );
+
+  let creationStatus = NOT_STARTED;
+  let creationActions = {};
+
+  const accountCreated = account !== null;
+  const fundedWithEthers = `${ethersBalance}` !== `${ZERO}`;
+  const fundedWithMana = `${manaBalance}` !== `${ZERO}`;
+  const marketplaceWasApproved = `${allowance}` != `${ZERO}`;
+
+  if (accountCreated) {
+    creationStatus = FUNDING_WITH_ETH;
+    creationActions[FUNDING_WITH_ETH] = null;
```

Comment:
Hmm, I think I personally would have modified the logic on the `My account` screen to use balances if it's an old account rather than putting fake actions in storage

Reply:
Partially because that screen will want access to that data even for new accounts pretty soon

Reply:
(This won’t gate merging this PR)

Reply:
I've just finished a refactoring on this component, I'll work to get account progress from blockchain now.

Reply:
Oh ok, nice!!

Reply:
Ok, great. Extract to: https://github.com/tasitlabs/tasit/issues/318
So, I'll do the test for an old account right now.

Reply:
Awesome, thanks!

Reply:
Resolving this